### PR TITLE
Sanitize host name in test input

### PR DESCRIPTION
**Pull request by vehre-x41**
Commit Message: Sanitize host name in fuzzer input
Additional Description:
Fuzzers tend to add characters into test input, that are not legal in hostnames. This patch fixes this for the header_parser_fuzz_test.
Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: test/fuzz/utility.h [hunk 1]
```diff
@@ -159,9 +159,18 @@ inline std::unique_ptr<TestStreamInfo> fromStreamInfo(const test::fuzz::StreamIn
...
+          ? (stream_info.address().address_case() ==
+                         envoy::config::core::v3::Address::AddressCase::kEnvoyInternalAddress &&
+                     stream_info.address().envoy_internal_address().address_name_specifier_case() ==
+                         envoy::config::core::v3::EnvoyInternalAddress::AddressNameSpecifierCase::
+                             kServerListenerName
+                 ? std::make_shared<Envoy::Network::Address::EnvoyInternalInstance>(
+                       replaceInvalidHostCharacters(
+                           stream_info.address().envoy_internal_address().server_listener_name()))
+                 : Envoy::Network::Address::resolveProtoAddress(stream_info.address()))
+          : Network::Utility::resolveUrl("tcp://10.0.0.1:443");
```
Comment (adisuissa) [inline test/fuzz/utility.h:173]: I think the logic is correct, but nested ternary-operators are hard to read/reason about.
Can you please use if-else here?

**Review by adisuissa — CHANGES_REQUESTED** [review]
Thanks for fixing this!
/wait

File: test/fuzz/utility.h (hunk 1 above)
Comment (vehre-x41) [reply test/fuzz/utility.h]: Changed. Is this how you like to see it?

**Review by adisuissa — APPROVED** [review]
LGTM, thanks for fixing this!

**Pull request merged**